### ISSUE TITLE
Using MBrace within an F# Azure Function stops the function from compiling

### ISSUE THREAD

**Issue by bruinbrown** [Azure/azure-webjobs-sdk-script#755]
Currently using Functions to deploy work to an MBrace cluster on a recurring schedule. As soon as a call is made in the function body to connect to an Azure cluster, the script is picked up as being changed, the message appears showing that the script is about to be compiled but then no more log messages appear to say the compilation completes and trying to execute the function fails.
#### Repro steps
1. Create a new F# function and add the MBrace.Azure NuGet package
2. Edit the function body to connect to a cluster i.e.

```
let Run(input:string) =
    let cluster = AzureCluster.Connect("", "")
    ()
```
1. Click Save on the function
#### Expected behavior

The function should be saved and compiled ready to be executed
#### Actual behavior

The compiler hangs and the function does not get compiled
#### Known workarounds

Creating an external DLL which holds all of the MBrace related calls in a single function works i.e. having a function in a DLL which is simply called RunMBrace which handles the connecting and deploying of the cloud workflow, but I did need to ILMerge the MBrace assemblies into this DLL.

Alternatively using MBrace with a C# function works as expected and calling AzureCluster.Connect won't cause the function compilation to hang

**Comment by lindydonna** [comment]
@fabiocav Thinks this is a NuGet reference issue

**Comment by christopheranderson** [comment]
Next unless we have a repro for more packages to prove this is a general issue. Not all F# issues need to be done by rc, as well.

**Comment by dsyme** [comment]
@bruinbrown  @lindydonna @fabiocav @christopheranderson - @sylvanc and I are actively looking into this bug this week. We think nearly all the existing F# issues boil down to the one same bug (not finding assemblies at runtime which have been resolved statically)

Suspected duplicate of #650, but we need to investigate further.

**Comment by sylvanc** [comment]
This issue can be closed.

The compiler no longer hangs. Nuget's transitive package resolution and this particular set of packages don't seem to be interacting quite right. If you explicitly include MBrace.Runtime 1.3.1 it works:

```json
{
  "frameworks": {
    "net46": {
      "dependencies": {
        "MBrace.Azure": "1.4.3",
        "MBrace.Runtime": "1.3.1"
      }
    }
  }
}
```

**Comment by sylvanc** [comment]
@fabiocav We're pretty sure there are some generic issues to do with non-isolation of project.json between Functions and between Function executions. We're pretty sure these aren't F# specific, as the same mechanism is used in C#.

When we added the 1.3.1 reference specifically, we had to `#r` the correct DLL to make things work (i.e. to force 1.3.1 instead of 1.3.0 to load), but once 1.3.1 was loaded, we could remove the `#r` and it still worked.

We know this stuff isn't easy, as we've had similar problems in our F# Azure Notebooks implementation :)
